sklearn/ensemble/_forest.py

```diff
@@ -494,7 +494,11 @@ def fit(self, X, y, sample_weight=None):
             # Collect newly grown trees
             self.estimators_.extend(trees)
 
-        if self.oob_score:
+        # Only perform OOB computation again if there are newly grown trees or
+        # if it was not previously calculated due to oob_score parameter being False
+        if self.oob_score and (
+            n_more_estimators > 0 or not hasattr(self, "oob_score_")
+        ):
```

Comment:
Could we isolate this change in another PR? It will require writing some new unit tests. Since the documentation would be mergeable as-it, it would be the best way to go.

Reply:
Hi @glemaitre, thank you for your comments. As suggested, I have updated my PR to remove the changes to the OOB computation and only kept the changes to the documentation.